pallets/anchors/src/benchmarking.rs

```diff
@@ -0,0 +1,119 @@
+use super::*;
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite, whitelisted_caller};
+use frame_support::traits::Get;
+use frame_system::RawOrigin;
+use sp_runtime::traits::Hash;
+use sp_runtime::DispatchError;
+
+const DOC_ROOT: [u8; 32] = [
+	238, 250, 118, 84, 35, 55, 212, 193, 69, 104, 25, 244, 240, 31, 54, 36, 85, 171, 12, 71, 247,
+	81, 74, 10, 127, 127, 185, 158, 253, 100, 206, 130,
+];
+
+const SIGNING_ROOT: [u8; 32] = [
+	63, 39, 76, 249, 122, 12, 22, 110, 110, 63, 161, 193, 10, 51, 83, 226, 96, 179, 203, 22, 42,
+	255, 135, 63, 160, 26, 73, 222, 175, 198, 94, 200,
+];
+
+const PROOF: [u8; 32] = [
+	192, 195, 141, 209, 99, 91, 39, 154, 243, 6, 188, 4, 144, 5, 89, 252, 52, 105, 112, 173, 143,
+	101, 65, 6, 191, 206, 210, 2, 176, 103, 161, 14,
+];
+
+fn day<T: From<u64>>(n: u64) -> T {
+	T::from(common::MILLISECS_PER_DAY * n + 1)
+}
+
+benchmarks! {
+	where_clause {
+		where
+		T: Config,
+		T::Hash: From<[u8; 32]>,
+		T::Moment: From<u64>,
+	}
+
+	pre_commit {
+		let caller = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, T::PreCommitDeposit::get());
+
+		let anchor_id = T::Hashing::hash_of(&0);
+
+	}: _(RawOrigin::Signed(caller), anchor_id, SIGNING_ROOT.into())
+	verify {
+		assert!(<PreCommits<T>>::get(anchor_id).is_some());
+	}
+
+	commit {
+		let caller = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, T::PreCommitDeposit::get());
+
+		let pre_image = T::Hashing::hash_of(&0);
+		let anchor_id = pre_image.using_encoded(T::Hashing::hash);
+
+		<Pallet<T>>::pre_commit(
+			RawOrigin::Signed(caller.clone()).into(),
+			anchor_id,
+			SIGNING_ROOT.into()
+		)?;
+
+	}: _(RawOrigin::Signed(caller.clone()), pre_image, DOC_ROOT.into(), PROOF.into(), day(1))
+	verify {
+		assert_eq!(T::Currency::free_balance(&caller), T::PreCommitDeposit::get());
+		assert!(<PreCommits<T>>::get(anchor_id).is_none());
+		assert!(<AnchorEvictDates<T>>::get(anchor_id).is_some());
+	}
+
+	evict_pre_commits {
+		let caller = whitelisted_caller();
+
+		let anchor_ids = (0..EVICT_PRE_COMMIT_LIST_SIZE)
+			.map(|i| {
+				T::Currency::make_free_balance_be(&caller, T::PreCommitDeposit::get());
+
+				let anchor_id = T::Hashing::hash_of(&i);
+
+				<Pallet<T>>::pre_commit(
+					RawOrigin::Signed(caller.clone()).into(),
+					anchor_id,
+					SIGNING_ROOT.into()
+				)?;
+
+				Ok(anchor_id)
+
+			})
+			.collect::<Result<Vec<_>, DispatchError>>()?
+			.try_into()
+			.expect("resulting BoundedVec is equal to EVICT_PRE_COMMIT_LIST_SIZE");
+
+		frame_system::Pallet::<T>::set_block_number(PRE_COMMIT_EXPIRATION_DURATION_BLOCKS.into());
+
+	}: _(RawOrigin::Signed(caller), anchor_ids)
+	verify {
+		assert_eq!(<PreCommits<T>>::iter_values().count(), 0);
+	}
+
+	evict_anchors {
```

Comment:
It is really difficult to accuracy benchmark this. It has two big parameters that affect the computation of this call. First, the `MAX_LOOP_IN_TX`, and second the number of days from the last `evict_anchors` call. The first one is handled. But the second one has no limit, and I do not know how to deal with it. Anyway, I don't think it is an issue, since this call is not idempotent. I mean, only the first call could have a big amount of computation, the next ones not.

Reply:
I think we should change the evict_anchor logic and simplify it to be called externally per anchor_id like we do with the evict_precommit. Lets chat more about it offline.